src/db/mysql/models.rs

```diff
@@ -327,43 +326,53 @@ impl MysqlDb {
         let user_id: u64 = bso.user_id.legacy_id;
         let timestamp = self.timestamp().as_i64();
 
-        // XXX: consider mysql ON DUPLICATE KEY UPDATE?
         self.conn.transaction(|| {
+            let payload = bso.payload.as_ref().map(Deref::deref).unwrap_or_default();
+            let sortindex = bso.sortindex;
+            let ttl = bso.ttl.map_or(DEFAULT_BSO_TTL, |ttl| ttl);
             let q = r#"
-                SELECT 1 as count FROM bso
-                WHERE user_id = ? AND collection_id = ? AND id = ?
+                INSERT INTO bso (user_id, collection_id, id, sortindex, payload, modified, expiry)
+                VALUES (?, ?, ?, ?, ?, ?, ?)
+                ON DUPLICATE KEY UPDATE sortindex = VALUES(sortindex)
```

Comment:
`sortindex` like the others is also optional.

We also need to handle the case of something issuing an empty update (all update fields being None). Though I mentioned earlier that we're not required to issue updates to the primary key fields (user_id/collection_id/id) let's include them anyway so we're *always* issuing at least something to the UPDATE clause.

It's kind of cheating but it'll keep the logic here simpler. (I just checked what the Python version does for empty updates and it does this, so this cheat works for me).